File: app/src/main/java/com/example/energo_monitoring/checks/data/db/ResultData.java
package com.example.energo_monitoring.checks.data.db;

import androidx.room.Embedded;
import androidx.room.Relation;


import com.example.energo_monitoring.checks.data.CheckLengthResult;
import com.example.energo_monitoring.checks.data.api.ClientInfo;
import com.example.energo_monitoring.checks.data.api.OrganizationInfo;
import com.example.energo_monitoring.checks.data.api.ProjectDescription;
import com.example.energo_monitoring.checks.data.devices.DeviceCounter;
import com.example.energo_monitoring.checks.data.devices.DeviceFlowTransducer;
import com.example.energo_monitoring.checks.data.devices.DevicePressureTransducer;
import com.example.energo_monitoring.checks.data.devices.DeviceTemperatureCounter;
import com.example.energo_monitoring.checks.data.devices.DeviceTemperatureTransducer;

import java.util.List;

public class ResultData {

    //@PrimaryKey
    @Embedded
    public OtherInfo otherInfo;

    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    public ClientInfo clientInfo;
    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    public OrganizationInfo organizationInfo;
    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    public ProjectDescription project;
    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceTemperatureCounter.class)
    public List<DeviceTemperatureCounter> deviceTemperatureCounters;
    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceFlowTransducer.class)
    public List<DeviceFlowTransducer> deviceFlowTransducers;
    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    public List<CheckLengthResult> flowTransducerLengths;
    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceTemperatureTransducer.class)
    public List<DeviceTemperatureTransducer> deviceTemperatureTransducers;
    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DevicePressureTransducer.class)
    public List<DevicePressureTransducer> devicePressureTransducers;
    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceCounter.class)
    public List<DeviceCounter> deviceCounters;

    public ResultData() {
    }

    public ResultData(int dataId, String projectPhotoBase64) {
        this.clientInfo.dataId = dataId;
        this.projectPhotoBase64 = projectPhotoBase64;
    }

    public String projectPhotoBase64;

}

File: app/src/main/java/com/example/energy_monitoring/checks/data/db/ResultData.kt
package com.example.energy_monitoring.checks.data.db

import androidx.room.Embedded
import androidx.room.Relation
import com.example.energy_monitoring.checks.data.api.ClientInfo
import com.example.energy_monitoring.checks.data.api.OrganizationInfo
import com.example.energy_monitoring.checks.data.api.ProjectDescription
import com.example.energy_monitoring.checks.data.devices.DeviceTemperatureCounter
import com.example.energy_monitoring.checks.data.devices.DeviceFlowTransducer
import com.example.energy_monitoring.checks.data.CheckLengthResult
import com.example.energy_monitoring.checks.data.devices.DeviceTemperatureTransducer
import com.example.energy_monitoring.checks.data.devices.DevicePressureTransducer
import com.example.energy_monitoring.checks.data.devices.DeviceCounter
import com.example.energy_monitoring.checks.data.files.FinalPhotoFile

class ResultData {
    @Embedded
    var otherInfo: OtherInfo? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    var clientInfo: ClientInfo? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    var organizationInfo: OrganizationInfo? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    var project: ProjectDescription? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceTemperatureCounter::class)
    var deviceTemperatureCounters: List<DeviceTemperatureCounter>? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceFlowTransducer::class)
    var deviceFlowTransducers: List<DeviceFlowTransducer>? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId")
    var flowTransducerLengths: List<CheckLengthResult>? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceTemperatureTransducer::class)
    var deviceTemperatureTransducers: List<DeviceTemperatureTransducer>? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DevicePressureTransducer::class)
    var devicePressureTransducers: List<DevicePressureTransducer>? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = DeviceCounter::class)
    var deviceCounters: List<DeviceCounter>? = null

    @Relation(entityColumn = "dataId", parentColumn = "dataId", entity = FinalPhotoFile::class)
    var finalPhotos = mutableListOf<FinalPhotoFile>()
}